design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,529 @@
+map the logical network name to pod interface name.
+
+The following table describes which information is accessible to each relevant KubeVirt component:
+
+| Component/has access to | VirtualMachineInstance | Pod   | Cluster-config |
+|-------------------------|------------------------|-------|----------------|
+| VMI controller          | True                   | True  | True           |
+| virt-handler            | True                   | False | True           |
+| virt-launcher           | True                   | False | False          |
+
+Cluster wide config passes from virt-handler to virt-launcher via their dedicated gRPC channel.
+
+The pod interfaces' naming scheme is currently independently calculated at each of the three components, based on input
+from VMI.Spec.
+
+The information about the pod's primary interface name should be propagated to all three components.
+
+## Solution
+
+### Inferring the primary interface name from Multus’ network-status annotation
+
+When KubeVirt is used in a cluster that has Multus as its default CNI, Multus adds the
+`k8s.v1.cni.cncf.io/network-status` annotation to the virt-launcher pod following its network attachment process.
+This annotation includes information about all the network interfaces attached to the pod.
+
+The VMI controller currently uses this annotation for:
+
+- Updating the VMI status interfaces section with the
+  `multus-status` [info-source](https://kubevirt.io/api-reference/v1.3.1/definitions.html#_v1_virtualmachineinstancenetworkinterface).
+- Creating the `kubevirt.io/network-info` annotation, which is the data source for DownwardAPI when using SR-IOV NICs
+  and certain network binding plugins.
+
+The `k8s.v1.cni.cncf.io/network-status` annotation shall be parsed to infer the default (primary) network interface
+instead of relying on a hard coded
+interface name (like eth0).
+
+> [!NOTE]
+> [ Kubernetes Network Custom Resource Definition De-facto Standard](https://github.com/k8snetworkplumbingwg/multi-net-spec/tree/master/v1.3)
+> states:
+>
+> "default" - This required key’s value (type boolean) shall indicate that this attachment is the result of the
+> cluster-wide default network.
+> Only one element in the Network Attachment Status Annotation list may have the "default" key set to true.
+>
+> "interface" - This optional key’s value (type string) shall contain the network interface name in the pod’s network
+> namespace
+> corresponding to the network attachment.
+
+VMI controller will look for the first entry to have `"default": true` and a non-empty `interface` field.
+If such entry exists - it would be considered as the pod's primary interface name and will override the default `eth0`.
+
+Example 1:
+
+When a CNI sets a custom name for the pod's primary interface we can expect the following:
+
+```json
+[
+  {
+    "name": "cluster-network",
+    "interface": "custom-iface",
```

Comment:
It is a non-goal, see line 31.
The scenarios we know of cannot have different names, no matter if it is an upgrade or just a simple migration.

If the name changes, it will just fail the migration, but we should not be in that position.

Reply:
Perfect, thank you @EdDev !

Reply:
@EdDev @vladikr please see https://github.com/kubevirt/user-guide/pull/845.